src/EnergyPlus/ChilledCeilingPanelSimple.cc

```diff
@@ -257,18 +257,17 @@ void GetCoolingPanelInput(EnergyPlusData &state)
     int IOStat;
     bool ErrorsFound(false); // If errors detected in input
     auto &cCurrentModuleObject = state.dataIPShortCut->cCurrentModuleObject;
-    state.dataChilledCeilingPanelSimple->NumCoolingPanels =
-        state.dataInputProcessing->inputProcessor->getNumObjectsFound(state, cCMO_CoolingPanel_Simple);
+    int NumCoolingPanels = state.dataInputProcessing->inputProcessor->getNumObjectsFound(state, cCMO_CoolingPanel_Simple);
 
     // Count total number of baseboard units
 
-    state.dataChilledCeilingPanelSimple->CoolingPanel.allocate(state.dataChilledCeilingPanelSimple->NumCoolingPanels);
-    state.dataChilledCeilingPanelSimple->CoolingPanelSysNumericFields.allocate(state.dataChilledCeilingPanelSimple->NumCoolingPanels);
-    state.dataChilledCeilingPanelSimple->CheckEquipName.allocate(state.dataChilledCeilingPanelSimple->NumCoolingPanels);
+    state.dataChilledCeilingPanelSimple->CoolingPanel.allocate(NumCoolingPanels);
+    state.dataChilledCeilingPanelSimple->CoolingPanelSysNumericFields.allocate(NumCoolingPanels);
+    state.dataChilledCeilingPanelSimple->CheckEquipName.allocate(NumCoolingPanels);
     state.dataChilledCeilingPanelSimple->CheckEquipName = true;
```

Comment:
Does this assignment operator set all elements of the array to `true`? That is a terrible idiom. Is this a Fortran legacy thing?